### ISSUE TITLE
TCP/53 and UDP/53 already taken on OSX/Linux

### ISSUE THREAD

**Issue by cytopia** [cytopia/devilbox#82]
@fe3dback moved from #63 

> I can't setup DB resolver. My steps:
> 
> Turn off my dnsmasq service, check host.
> Replace port to default 53 in HOST_PORT_BIND env (before 54)
> add nameserver 127.0.0.1 in /etc/resolv.conf
> Restart my network service
> Restart docker-compose
> In devilbox /vhosts.php page, all my projects have label "No Host DNS record found".
> 
> So, now i returned to my dnsmasq service.

---

@jhabdas moved from #63 

> [TCP and UDP ports used by Apple software products](https://support.apple.com/en-us/HT202944) states 53 is taken but does not mention 54.
>
> My out of the devilbox experience following `.env` config was:
>
> ```
> devilbox master ❯ docker-compose up
> ERROR: Invalid published port: 53/udp
> devilbox master ❯ docker-compose up
> ERROR: Invalid published port: 54/udp
> ```
>
> ---
> 
> Running a UDP scan of 53/54:
> 
>     sudo nmap -sU -p 53,54 localhost
> 
> Results in:
> 
> ```
> PORT   STATE  SERVICE
> 53/udp closed domain
> 54/udp closed xns-ch
> ```
> 
> ---
> 
> Ran `scutil --dns` to check resolvers. Lots of output. Not dumping here...
> 
> ---
> 
> IIRC the solution I've used for this in the past was `consul` or similar. Here's a blog talking more about Consul's service discovery and DNS features: http://blog.scottlowe.org/2015/02/06/quick-intro-to-consul/.
> 
> I need to move onto other work. I'll let someone else do the honors of opening a new issue. Tool looks promising. Looking forward to using it on macOS.

**Comment by cytopia** [comment]
I will move TCP/53 and UDP/53 both to 1053 by default and add some documentation for how to change and integrate into Linux and OSX

**Comment by boligolov** [comment]
I had same problem in Ubuntu:

> ERROR: Invalid published port: 53/udp

Commenting the line resolved the problem ;)

**Comment by cytopia** [comment]
@boligolov the problem was a wrong assignment for UDP ports. I found this explanation to it.

https://github.com/docker/compose/issues/1334

As you can see from the above commit, it will be fixed in the upcoming version.

**Comment by jhabdas** [comment]
![angelfives](https://user-images.githubusercontent.com/440298/27353988-a1e90192-5638-11e7-8f43-329a3f83900d.gif)

**Comment by cytopia** [comment]
resolved in latest tagged version.

**Comment by jhabdas** [comment]
Bind port updated. Pulling layers. I'm on a pretty slow connection so don't hold your breath.

**Comment by jhabdas** [comment]
Install works now. Composer is up. Bunch of verbose output from Maria. Now I just gotta figure out how to use this thing. @cytopia thoughts on moving the video tutorials over to Vimeo to nix the trackers and improve security?

**Comment by cytopia** [comment]
@jhabdas

> Now I just gotta figure out how to use this thing

Just added a huge documentation. Let me know if anything is missing.

> thoughts on moving the video tutorials over to Vimeo to nix the trackers and improve security?

`nix the trackers` ? And why does it improve security?

**Comment by jhabdas** [comment]
Our dear friend google uses double-click to sell PII to third party advertisers. And I learned recently Vimeo won't allow videos on HTTP to play over insecure connections. [Example here](https://hackcabin.com/post/become-digital-nomad-bali/). Scroll below the fold. Do you see the video?

Awesome. Parsing through the docs now. Mad props for being thorough. ✊

**Comment by jhabdas** [comment]
Initially could not get outside the php box for wordpress setup after composing. running docker update to get new layers.

You may be interested in:
https://github.com/jlk/wordpress-swarm
https://github.com/TrafeX/docker-wordpress

**Comment by cytopia** [comment]
You can use provided `update.sh` or `docker-compose pull`. Thx for the links

**Comment by cytopia** [comment]
Btw, there is an example setup for wordpress:

https://github.com/cytopia/devilbox/blob/master/docs/Examples.md#6-setup-wordpress

**Comment by jhabdas** [comment]
I found that quickly. Your docs are very well laid out and easy to navigate. 👌

**Comment by jhabdas** [comment]
While we're sharing. I'd be a fool if I didn't send this your way: https://serifandsemaphore.io/how-to-host-wordpress-like-a-boss-b5993fcfbd8e

**Comment by jhabdas** [comment]
Probably doing something stupid. But I did a `.docker-update.sh`, pulled down layers. Then did a `docker-compose down` followed by a `docker-compose up` daemonized, shelled into the box as non-root and can't get an outside line for some reason:

```
devilbox@php-7.1.6 in /shared/httpd/my-wp $ ping habd.as
ping: habd.as: Name or service not known
```

Will dig more soon. Everything else is looking super sweet. I'm very captivated by this tool.

**Comment by cytopia** [comment]
@jhabdas could you open up a separate issue for this problem. Others won;t see this in a closed ticket.

**Comment by jhabdas** [comment]
I will only once I've done my due diligence first.

**Comment by boligolov** [comment]
@cytopia thank you! :)
